cmake/compiler/gcc/target.cmake

```diff
@@ -14,6 +14,29 @@ find_program(CMAKE_READELF    ${CROSS_COMPILE}readelf PATH ${TOOLCHAIN_HOME} NO_
 find_program(CMAKE_GDB        ${CROSS_COMPILE}gdb     PATH ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
 find_program(CMAKE_NM         ${CROSS_COMPILE}nm      PATH ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
 
+# x86_64 should pick up a proper cross compiler if one is provided,
+# but falling back to using the host toolchain is a very sane behavior
+# too.
+if(CONFIG_X86_64)
+  if(CMAKE_C_COMPILER STREQUAL CMAKE_C_COMPILER-NOTFOUND)
```

Comment:
@andyross did you actually test this fallback? I can run it and make it work but only after some serious hacking because the same check in `compiler/gcc/generic.cmake` runs first and _aborts cmake first_ before this target.cmake file is even looked at.

In the cover letter of this PR you also wrote:
> The architecture has to be built on a Linux host with ZEPHYR_TOOLCHAIN_VARIANT=host.

ZEPHYR_TOOLCHAIN_VARIANT=host means no CROSS_COMPILE and none of this `compiler/gcc/target.cmake` code is used at all! (`compiler/host-gcc/target.cmake` is used instead)
Were you testing **both** ZEPHYR_TOOLCHAIN_VARIANT=host and cross-compile?

Reply:
This fallback has just been removed: https://github.com/zephyrproject-rtos/zephyr/commit/33ca224ffdda90